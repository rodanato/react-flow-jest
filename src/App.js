// @flow

import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Box from './box.component';

type Props = {};

type State = {
  list: string[]
};

class App extends Component<Props, State> {
  state = {
    list: []
  };

  loadItemstoList(list: string[]) {
    let newList = list;

    newList.push('uno');
    newList.push('223');

    return newList;
  }

  componentDidMount() {
    const list = this.loadItemstoList(this.state.list);

    this.setState({list: list});
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>

        {this.state.list.map((item, i) => <h2 key={i}>{item}</h2>)}

        <Box/>
      </div>
    );
  }
}

export default App;
