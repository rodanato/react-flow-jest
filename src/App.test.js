
import React from 'react';
import App from './App';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Box from './box.component';


configure({ adapter: new Adapter() });

describe('Component: App', () => {
  const items = ['Learn react', 'rest', 'go out'];
  const comp = shallow(<App />);

  it('should match its empty snapshot', () => {
    expect(comp).toMatchSnapshot();
  });

  it('should render a h1 title', () => {
    expect(comp.find('h1').length).toEqual(1);
  });

  it('should render the Box Component', () => {
    expect(comp.containsMatchingElement(<Box />)).toEqual(true);
  });

  // it('should add an item based on the value in the state', () => {
  //   const component = shallow(<App />);
  //   const preventDefault = jest.fn();
  //   component.setState({
  //     items
  //   });
  //   component.find('form').simulate('submit', { preventDefault });
  //   expect(toJson(component)).toMatchSnapshot();
  //   expect(preventDefault).toBeCalled();
  // });
});

