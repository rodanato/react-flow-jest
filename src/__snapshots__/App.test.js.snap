// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should match its empty snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <h2>
    uno
  </h2>
  <h2>
    223
  </h2>
</div>
`;
